pokedex/share.lua

```diff
@@ -79,12 +78,19 @@ local function decode_status(pokemon)
 	pokemon.statuses = new
 end
 
+local function serialize_pokemon(pokemon)
+	decode_status(pokemon)
+	return ljson.encode(pokemon)
+end
+
+function M.generate_qr(id)
+	local pokemon = storage.get_copy(id)
```

Comment:
Why do you need a copy here?

Reply:
I don't, but that's the way you currently get the pokemon data. As discussed this will be changed in the storage/pokemon refactor